crates/tasks/src/lib.rs

```diff
@@ -271,7 +272,11 @@ impl TaskExecutor {
     {
         let on_shutdown = self.on_shutdown.clone();
 
+        // Wrap the original future to increment the finished tasks counter upon completion
+        let metrics = self.metrics.clone();
         let task = async move {
+            // Create an instance of IncCounterOnDrop with the counter to increment
+            let _inc_counter_on_drop = IncCounterOnDrop::new(metrics.finished_regular_tasks);
```

Comment:
we only need to clone one counter so the `IncCounterOnDrop` can be initialized outside

Reply:
that's why you need to move it inside.

for example call drop on it manually

Reply:
Oh I think I misunderstood your previous review.

So you mean to:

- clone just the specific metrics counter needed.
- may use a drop manually (?). Is this needed? As 'IncCounterOnDrop' is in the same scope of the async block, it should be ok.

Reply:
> clone just the specific metrics counter needed.

yes

> may use a drop manually (?). Is this needed? As 'IncCounterOnDrop' is in the same scope of the async block, it should be ok.

you must ensure the value is actually moved inside the closure.